### ISSUE TITLE
use case for .clearCache()

### ISSUE THREAD

**Issue by kevinGodell** [vot/ffbinaries-node#12]
I was testing deployment with your module because I am including it with my [rtsp-cam-tester](https://github.com/kevinGodell/rtsp-cam-tester/releases) node app. I was testing various configurations and came across an error, which could very well be my fault. When my app loads, I called .clearCache() which did not throw an error. Later, when my app did some checking to see if ffmpeg is available and initiates a download via your module, it throws an error 
```
internal/streams/legacy.js:59
      throw er; // Unhandled stream error in pipe.
      ^

Error: ENOENT: no such file or directory, open '/Users/kevinGodell/.ffbinaries-cache/ffmpeg-3.4-osx-64.zip.part'
```
So it seems that the cache was cleared, but when trying to download a fresh copy, it tried to look for the cache and did not find it. Otherwise, great module. Thanks. I don't really need this fixed. I just came across it when doing some testing that would not occur in production for me.

P.S. The main reason I am using your module vs the other ffmpeg modules is that yours gives me the control I needed to install only if i needed to.  I found it impossible to package ffmpeg with my app when using "pkg."

**Comment by vot** [comment]
Hi Kevin, thanks for reporting this.
I suspect it might have something to do with `clearCache` method removing the whole `.ffbinaries-cache` directory instead of just its contents. The cache directory gets created early on in the library - when removing the whole directory it doesn't currently recreate that directory.

FWIW I believe clearing cache might be redundant as version 1.0.3 introduced some changes that prevent cache corruption through incomplete files (this commit: https://github.com/vot/ffbinaries-node/commit/606f6dd9009a5dd2a51c53711837acde79785138).

PS Glad to hear you find this module useful. If there's any way to make it fit better into your workflow let me know and we can look into that. The whole point is to make getting these binaries easier :)

**Comment by vot** [comment]
Hi @kevinGodell,

I just published v1.0.4 to npm - can you please check if that resolves the issue?
The only difference is that clearCache will only remove contents of the directory rather than the directory itself.

**Comment by kevinGodell** [comment]
Thanks. I was just working on the fix, but you know your code better than me. That fix may make one of your tests fail that checks that the directory was removed. I will test and let you know.

**Comment by vot** [comment]
I actually meant to fix this a while back as it was the reason `clearCache` was run as the last test. 

I broke that test when changing the implementation. Fortunately all it has to do is verify the folder exists and it contains no zip files so it was reasonably quick. I also moved it to be executed earlier to ensure the problem you ran into doesn't happen again.
It all should be a bit more robust now :)

PS quite an interesting project you're building, good luck!
